ONSdigital/response-operations-ui
# Fix generate new enrolment code link

# What and why?

This fixes a bug in the "Generate new enrolment code" hyperlink. This link is currently only formed correctly if registered respondents exist for the RU/survey. If no respondents exist the link is malformed and doesn't work. Respondents who have not verified their accounts within 80 hours will be deleted leaving no active UAC codes and no way for the business to generate new codes.

The link should be formed and presented if an active IAC code doesn't already exist, even if no respondents are registered for the business.

# How to Test

- Create a collection exercise through to it going live.
- Get the IAC from the `/reporting-units/{ru_ref}/surveys/{survey_id}` page.
- Register a user with the code but don't verify them.
- Mouse over (but don't click) the "Generate new enrolment code" link and check its correct in the `/reporting-units/{ru_ref}/surveys/{survey_id}` page  (see the unit test in this PR for the expected correct format). You should see the respondent showing as pending verification in the page.
- Delete the party `respondent` record (inc its `business_respondent`, `enrolment` and `pending_enrolment` records). You can do this manually via a Postgres client or use the delete account scheduling jobs if you hack at the dates to make the account creation date more than 80 hours (this is what surfaced the production bug)
- Refresh the `/reporting-units/{ru_ref}/surveys/{survey_id}` page and check there are no respondents showing in the page.
- Click the "Generate new enrolment code" link in the `/reporting-units/{ru_ref}/surveys/{survey_id}` page  (see the unit test in this PR for the expected correct format). A new enrolment code should be presented, and this should now show in the `/reporting-units/{ru_ref}/surveys/{survey_id}` page instead of a "Generate new enrolment code" link.
- Register a new user with this new enrolment code.

https://jira.ons.gov.uk/browse/RAS-1149
